### DISCUSSION TITLE
Trying to duplicate my TV top LED row to another separated strip to the ceiling. It works, but feels clunky, I think I'm not doing it right

### DISCUSSION THREAD

**Discussion by x130844** [awawa-dev/HyperHDR#727] [Q&A]
Hi, I tried different setups, I was able to get it to work with hyperion and RPI direct gpio. I know, not supported, but I didn't know that then :)    Since then, I'm converting to hyperHDR, and hyperspi32_esp32 (which functionally speaking works). 

What I'm trying to accomplish:
1 LED strip around the TV.  (4 corners), using USB video grabber. 
 Here's the catch,   1 extra LED strip, longer than the top row strip. and place it way above the cabinets. 
I got it to work (though, not reliably), with hyperion , created 2 instances,  gpio pwm pin 18/0, and pin19/1
First instance has the whole TV frame setup.  USB video grabber.  (nothing special) 
2nd instance is defined as top row only to the LED config,  but also pointing to the same USB video grabber. 

It worked for a few hours, then would disconnect, it would switch position, reverse orientation etc, etc, but when it was working, it was great.    
  Question 1)   it feels wrong to set 2 instances to the same video capture.  Am I creating issues doing this or is this the proper way to do it? 
Question 2) switching to HyperHDR > RPI4 > hyperspi32_esp32.     , For the ceiling strip,  would I need a 2nd esp32 connected to the RPI4?     or can I use the current esp32 and use another pin out to control the data of the 2nd strip? 
I saw in the doc that I could do split segments for doubling, but this is not my case scenario here. 
If I was to have a 2nd strip LED data pin out on the esp32 , how would I tell HyperHDR to point to that?   

Am I missing something?   I saw in HyperHDR some group (horizontal/vertical) but I couldn't make it sense and I can't seem to find any doc about that.  

Help?

**Comment by awawa-dev** [comment]
Hi
Although this is an undocumented functionality, it assumes that:
- the first segment will be equal or longer than the second
- we do not use the SECOND_SEGMENT_REVERSED option

then multi-segment will work for you in such scenario. In HyperHDR you will have one very long strip at your disposal and one instance will be enough. A separate issue is how to configure it, but I suspect that the easiest way will be to configure the first (larger) setup, copy the json with the LED definition to the editor, clean it and configure the top bar in HyperHDR again, then merge the new configuration in the editor at the end and finally copy everything back to HyperHDR.

This configuration approach should also work even with the solution you currently have, as long as the end of the first strip connects to the beginning of the top one so you can test it now. With multi-segment, each LED system has a separate data line but since the signal lines will probably be quite long, level shifter is a must (or eg. Adafruit Scorpio for HyperSerialPico)

**Comment by x130844** [comment]


**Reply by awawa-dev** [reply]
Using multiple instances is of course ok, especially if you want to handle the second bar via wled. Maybe I explained it wrong, but it seems to me that it is simpler to use one instance where both strips are connected to separate signal lines on RP2040 with hyperserialpico/hyperspi. Using multiple instances may cause them run a bit o sync since the second instance for top led strip would use wled/WiFi driver.

**Reply by awawa-dev** [reply]
BTW, if you already have two instances configured, then you almost have a ready-made solution for one instance: you just need to connect the geometries. If you export your HyperHDR configuration (general->backup), I will show you what I mean.

**Comment by awawa-dev** [comment]
The ceiling in your configuration is at the bottom of the TV screen. Anyway here's a config:
[HyperHDR-20.0.0.0beta1-Backup-2024-01-14.json](https://github.com/awawa-dev/HyperHDR/files/13933195/HyperHDR-20.0.0.0beta1-Backup-2024-01-14.json)
The merged element begins before:
``` 
                        ......
			"hyperhdr_instance": 2,
			"type": "leds"
		},
```

**Reply by awawa-dev** [reply]
> Is what you did, only achievable via json manipulation only? or is it possible to accomplish it with the GUI as well? If from json only, that would explain why my GUI view doesn't make much sense.

This is simply a text editor operation on the exported configuration.
If you have two instances of 0 and 1, you have two sections which defines LEDs
```                  
 {
"config": "[{A1},{A2}...{AN}]"
"hyperhdr_instance": 0,
"type": "leds"
},
....
 {
"config": "[{B1},{B2}...{BN}]"
"hyperhdr_instance": 1,
"type": "leds"
},
```
so the final new instance for example 2 (must be created earlier) will have
```
...
 {
"config": "[{A1},{A2}...{AN},{B1},{B2}...{BN}]"
"hyperhdr_instance": 2,
"type": "leds"
},
```

if you have such a merged configuration and you want to delete previuos two instance, you can delete your current HyperHDR configuration in HyperHDR home directory (path in the logs, first lines), run HyperHDR and export a new clean configuration, then paste/replace the prepared "config" array in the text editor. Then import it again.

> PS: hyperHDR is WAY more stable than Hyperion. I don't know how Hyperion does it, but it crashes, hangs so many times, I'd almost unusable when trying to set it up. Your fork is way better. congrats.
Oh, where is the brightness gain option? I couldn't find it in hyperhdr? brightness option maxes out at 100%. the output appears less bright.

Thanks. I dont know what brightness you want to increase: LED in processing tab (try enable/disable classic HyperHDR configuration) , LED driver (ed. WLED has a power limiter) or video.
